### DISCUSSION TITLE
What is the role of XAML Islands in WinUI 3?

### DISCUSSION THREAD

**Discussion by memeplex** [microsoft/WindowsAppSDK#465] [Q&A]
If find this assertion in https://docs.microsoft.com/en-us/windows/apps/winui/ a bit confusing:

> WinUI [3] is the path forward for all Windows apps—you can use it as the UI layer on your native UWP or Win32 app, or you can gradually modernize your desktop app, piece by piece, with XAML Islands.

Does it mean that in the piece-by-piece scenario it will still have this underlying architecture:

![image](https://user-images.githubusercontent.com/2845433/108584757-9624ab80-7322-11eb-91d7-c98c73c4c750.png)

or just this one:

![image](https://user-images.githubusercontent.com/2845433/108584761-a046aa00-7322-11eb-997d-3b21ac831a70.png)

Maybe the real problem is that it's not clear to me whether XAML Islands are a way of bridging Win32 to UWP or more of a way of composing different kinds of components. Maybe it's both in WinUI 2 but just the second one in WinUI 3. IDK.

**Comment by andrewleader** [comment]
@ryandemopoulos can you answer this? Thanks!

**Reply by ryandemopoulos** [reply]
Tagging @marb2000 to answer

**Reply by marb2000** [reply]
Graphics are always confusing,  even more, the component layer ones.  They are not accurate and oversimplify things.  Let's see if some words bring a more clarity,

At glance:

**WinUI 3 in Desktop allows you to create a new Win32 app using the WinUI 3 as a standalone UI Platform**. You can't mix WinUI 3 content with other UI Platforms like WPF, WinForms, even Win32 Common Controls.  Fortunately, there are some controls in WinUI 3 that allow a WinUI 3 app to host other UI technologies like the WebView2 (for web content) and SwapChainPanel (for DirectX).

On the other hand, **WinUI 3's XAML Islands will allow you to modernize UI parts of your existing app**. You will be able to host WinUI 3 content in UI Frameworks based on HWNDs (WPF, WinForms, MFC).  For example, you can place WinUI controls inside a .NET 5 WinForms using the WinForm's WindowXamlHost control. Still, you can't host or place WinForms/WPF controls in areas where a WinUI 3 control is rendered. 

>**Note:** Today you can't mix in the same app WinUI 2/UWP XAML and WinUI 3 XAML. And this valid for Islands as well. You can't have a WPF app using WinUI 2 (using system XAML Islands) and WinUI 3 (using WinUI 3 XAML Islands) in the same process.
